K2Bridge/Visitors/Aggregations/MetricAggregationVisitor.cs

```diff
@@ -108,10 +127,63 @@ public void Visit(PercentileAggregation percentileAggregation)
             var valuesForColumnNames = string.Join(sep, percentileAggregation.Percents.ToList().Select(item => $"{item:0.0}"));
             var valuesForOperator = string.Join(',', percentileAggregation.Percents);
 
+            var metadata = new List<string>();
+            metadata.Add(percentileAggregation.Key);
+            metadata.Add(AggregationsConstants.Percentile);
+            metadata.Add(valuesForColumnNames);
+            metadata.Add(percentileAggregation.Keyed.ToString());
+
             // We don't use EncodeKustoField on this key because it contains a '.' but isn't dynamic
             // Example: ['A%percentile%25.0%50.0%99.0%False']=percentiles_array(fieldA, 25,50,99)']
-            var key = $"['{percentileAggregation.Key}{sep}percentile{sep}{valuesForColumnNames}{sep}{percentileAggregation.Keyed}']";
-            percentileAggregation.KustoQL = $"{key}={KustoQLOperators.PercentilesArray}({EncodeKustoField(percentileAggregation)}, {valuesForOperator})";
+            var keyWithMetadata = $"['{string.Join(AggregationsConstants.MetadataSeparator, metadata)}']";
+            percentileAggregation.KustoQL = $"{keyWithMetadata}={KustoQLOperators.PercentilesArray}({EncodeKustoField(percentileAggregation)}, {valuesForOperator})";
+
+            VisitedMetrics.Add(keyWithMetadata);
+        }
+
+        /// <inheritdoc/>
+        public void Visit(TopHitsAggregation topHitsAggregation)
+        {
+            Ensure.IsNotNull(topHitsAggregation, nameof(topHitsAggregation));
+            EnsureClause.StringIsNotNullOrEmpty(topHitsAggregation.Field, topHitsAggregation.Field, ExceptionMessage);
+
+            // top 2 by timestamp asc
+            var sort = topHitsAggregation.Sort.First();
+            var topHitsExpression = $"{KustoQLOperators.Top} {topHitsAggregation.Size} by {EncodeKustoField(sort.FieldName)} {sort.Order}";
+
+            // ['4']=pack('field', AvgTicketPrice, 'order', timestamp)
+            var data = new Dictionary<string, string>();
+            data.Add($"'{AggregationsConstants.SourceField}'", $"'{topHitsAggregation.Field}'");
+            data.Add($"'{AggregationsConstants.SourceValue}'", EncodeKustoField(topHitsAggregation.Field));
```

Comment:
For EncodeKustoField here and in the next line, add wrapDynamic=true. 
It should be true whenever we need the type of the dynamic field to be correct.
Right now it doesn't work with dynamic fields (test it)

Reply:
Fixed.